Duplicati/Library/Backend/GoogleServices/GoogleCommon.cs

```diff
@@ -195,26 +219,36 @@ private static async Task<T> ChunkedUploadAsync<T>(OAuthHelper oauth, string upl
                                 throw new Exception(string.Format("Upload succeeded prematurely. Uploaded: {0}, total size: {1}", offset, stream.Length));
 
                             //Verify that the response is also valid
-                            var res = oauth.ReadJSONResponse<T>(resp);
+                            T res = oauth.ReadJSONResponse<T>(resp);
                             if (res == null)
                                 throw new Exception(string.Format("Upload succeeded, but no data was returned, status code: {0}", code));
 
                             return res;
                         }
                         else
                         {
-                            throw new WebException(string.Format("Unexpected status code: {0}", code), null, WebExceptionStatus.ServerProtocolViolation, resp);
+                            /*
+                             * Grab the error detail from the response stream
+                             */
+                            using (Stream exstream = resp.GetResponseStream())

```

Comment:
I don't think you can guarantee there to be a response stream present? If the error is "Network disconnected" there will be no response stream and the user will get an exception here.